src/Sulu/Bundle/MediaBundle/Resources/js/views/MediaOverview/MediaOverview.js

```diff
@@ -1,75 +1,110 @@
+import {withToolbar, Datagrid, DatagridStore} from 'sulu-admin-bundle/containers';
 import type {ViewProps} from 'sulu-admin-bundle/containers';
-import {Masonry} from 'sulu-admin-bundle/components';
-import MediaCard from '../../components/MediaCard';
+
+const COLLECTION_ROUTE = 'sulu_media.overview';
+const COLLECTIONS_RESSOURCE_KEY = 'collections';
 
 @observer
 class MediaOverview extends React.PureComponent<ViewProps> {
-    medias: Array<*> = [
-        { id: 1, size: '260/350', title: 'This is a boring title', meta: 'bo and ring' },
-        { id: 2, size: '260/260', title: 'Is this one better?', meta: 'No' },
-        { id: 3, size: '260/300', title: 'But now!', meta: 'Hmm, not sure' },
-        { id: 4, size: '260/260', title: 'You want to have a fight?', meta: 'Come at me!' },
-        { id: 5, size: '260/380', title: 'LOL', meta: 'Yea, I thought so' },
-        { id: 6, size: '260/200', title: 'Now back to the Masonry', meta: ':)' },
-        { id: 7, size: '260/400', title: 'This is an image', meta: 'You are so smart' },
-        { id: 8, size: '260/180', title: 'This image has meta info', meta: 'No' },
-        { id: 9, size: '260/250', title: 'Dude, cmon', meta: 'NO' },
-        { id: 10, size: '260/200', title: 'Pls, you are embarrassing me', meta: 'Ugh, ok' },
-        { id: 11, size: '260/150', title: 'An image', meta: 'image/png, 3,2 MB' },
-    ];
+    page: observable = observable();
+    locale: observable = observable();
+    @observable title: string;
+    @observable parentId: ?string | number;
+    @observable collectionStore: DatagridStore;
+    @observable collectionId: string | number;
+    disposer: () => void;
+
+    componentWillMount() {
+        const {router} = this.props;
 
-    @observable selectedMediaIds: Array<string | number> = [];
+        router.bindQuery('page', this.page, '1');
+        router.bindQuery('locale', this.locale);
 
-    @action handleMediaCardSelectionChange = (id: string | number, selected: boolean) => {
-        if (selected) {
-            this.selectedMediaIds.push(id);
-        } else {
-            this.selectedMediaIds = this.selectedMediaIds.filter((selectedId) => selectedId !== id);
+        this.disposer = autorun(this.load);
+    }
+
+    componentWillUnmount() {
+        const {router} = this.props;
+        this.disposer();
+        router.unbindQuery('locale');
+        router.unbindQuery('page');
+        this.collectionStore.destroy();
+    }
+
+    load = () => {
+        const {router} = this.props;
+        const {
+            attributes: {
+                id,
+            },
+        } = router;
+
+        if (id) {
+            this.loadCollectionInfo(id);
         }
+
+        this.createCollectionStore(id);
     };
 
-    isSelected = (id: string | number) => {
-        return this.selectedMediaIds.includes(id);
+    loadCollectionInfo(collectionId) {
+        return ResourceRequester.get(COLLECTIONS_RESSOURCE_KEY, collectionId, {
```

Comment:
@chirimoya WDYT, should we just have `ResourceRequester` calls like this one in here?